DOCUMENTATION.md

```diff
@@ -0,0 +1,448 @@
+By default, all packages with the exception of `mas` and `mas-dev`, use the `debug` and `release` signing configurations.
+
+Using the example config above, let's go over some use cases to better understand the exceptions.
+
+**Use Case 1:**
+
+```
+$ cordova build electron --debug
+```
+
+The command above will:
+* Generate a `dmg` build and `mas-dev` build using the `debug` signing configurations.
+* Ignore the `mas` target package.
+
+*Use Case 2:*
+
+```
+$ cordova build electron --release
+```
+
+The command above will:
+* Generate a `dmg` build using the `release` config.
+* Generate a `mas` build using the `store` config.
+* Ignore the `mas-dev` target package.
+
+
+### Windows Signing
+
+The signing information is comprised of two types. (`debug`, `release`). Each section has the following properties:
+
+| key                       | description                                                                                                                                          |
+| ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+| certificateFile           | String path to the certificate file.                                                                                                                 |
+| certificatePassword       | String value of the certificate file's password.<br /><br />**Alternative**: The password can be set on the environment variable `CSC_KEY_PASSWORD`. |
+| certificateSubjectName    | String value of the signing certificate's subject.<br /><br />:warning: Required for EV Code Signing and requires Windows                            |
+| certificateSha1           | String value of the SHA1 hash of the signing certificate.<br /><br />:warning: Requires Windows                                                      |
+| signingHashAlgorithms     | Collection of singing algorithms to be used. (`sha1`, `sha256`)<br /><br />:warning: AppX builds only support `sha256`                               |
+| additionalCertificateFile | String path to the additional certificate files.                                                                                                     |
+
+**Example Config:**
+
+```
+{
+  "electron": {
+    "windows": {
+      "package": [
+        "nsis"
+      ],
+      "signing": {
+        "release": {
+          "certificateFile": "path_to_files",
+          "certificatePassword": "password"
+        }
+      }
+    }
+  }
+}
+```
+
+### Linux Signing
+
+There are not signing requirements for Linux builds.
+
+## Plugins
+
+All browser-based plugins are usable with the Electron platform.
+
+When adding a plugin, if the plugin supports both Electron and Browser, the Electron portion will be used. If the plugin contains the browser platform, but missing Electron, the browser platform will be used.
```

Comment:
```suggestion
When adding a plugin, if the plugin supports both the `electron` and `browser` platform, the `electron` portion will be used. If the plugin misses `electron`, but contains the `browser` implementation, it will fall back on the `browser` implementation.
```